stream/src/main/scala/org/apache/pekko/stream/scaladsl/Flow.scala

```diff
@@ -1141,7 +1140,18 @@ trait FlowOps[+Out, +Mat] {
    *
    * @see [[#mapAsync]]
    */
-  def mapAsyncUnordered[T](parallelism: Int)(f: Out => Future[T]): Repr[T] = via(MapAsyncUnordered(parallelism, f))
+  def mapAsyncUnordered[T](parallelism: Int)(f: Out => Future[T]): Repr[T] =
+    unsafeTransformUnordered[T](parallelism) { (out, emitter) =>
+      val future = f(out)
+      future.value match {
+        case Some(elem) => emitter.asInstanceOf[UnsafeEmitter[T]].handleNow(elem)
+        case None       => future.onComplete(emitter.handle)(pekko.dispatch.ExecutionContexts.parasitic)
+      }
+    }
```

Comment:
this change is a **demo** of how this method can be used

Reply:
I tried with another idea, better in Scala, but will make the function type in Java much complex.
```scala
    unsafeTransformUnordered[T](parallelism) { implicit emitter => out =>
      import EmitterOps._
      import EmitterUnsafeOps._
      val future = f(out)
      future.value match {
        case Some(elem) => handleNow(elem)
        case None       => future.onComplete(handle)(pekko.dispatch.ExecutionContexts.parasitic)
      }
    }
```